src/main/java/SoloTech/StockFlow/order/service/OrderService.java

```diff
@@ -24,18 +31,41 @@
 public class OrderService {
 
     private final OrderRepository orderRepository;
+    private final ProductService productService;
+    private final StockService stockService;
+    private final PaymentService paymentService;
     private final ObjectMapper mapper;
 
     @Transactional
     public Order createOrder(OrderDto dto) {
+        // 1. 상품 조회
+        Product product = productService.getProduct(dto.getProductId());
+        if (product == null) {
+            throw new RuntimeException("Product not found: " + dto.getProductId());
+        }
+
+        // 2. 재고 확인
+        Stock stock = stockService.getStock(dto.getProductId());
+        if (stock.getStock() < dto.getQuantity()) {
+            throw new RuntimeException("Insufficient stock for product: " + dto.getProductId());
+        }
+
+        // 3. 결제 처리
+        String initialPaymentStatus = "PENDING";
```

Comment:
```suggestion
        String initialPaymentStatus = "PENDING";
```
현재는 결제가 실패한다면 주문이 생성되지 않는데요

PENDING 이라는 상태값이 필요할까요?

Reply:
결제 생성하는 부분에서 바꿔주려고 생각했습니다.